src/Identity/AwsCredentialIdentity.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Aws\Identity;
+
+/**
+ * Denotes the use of standard AWS credentials.
+ *
+ * @internal
+ */
+class AwsCredentialIdentity implements IdentityInterface
+{
+    public function getExpiration() {}
```

Comment:
I don't think it's necessary. I've mentioned before that we're using these for Identification purposes only since Credentials, Token, etc. already implement the desired functionality.  I've marked them as internal, but maybe I need to be a bit more explicit about what they are used for?

Reply:
I'd both be explicit about that in docs and make these abstract classes/methods. These classes could technically be instantiated and used, however invalidly, while satisfying the defined contract. Marking them abstract doesn't harm functionality of the already implemented pieces, as they've implemented it, and you still get accurate type checking via `instanceof`.

Reply:
Puzzled as to why I pushed back on this initially, but I think my wires were crossed when thinking about how this would work in the implementation/inheritance hierarchy.  Updated.